react-hook/cdn-implementation/index.html

```diff
@@ -0,0 +1,164 @@
+            key: i,
+            className: i % 10 === 0 ? "large tick" : "tick",
+            style: {
+              transform: `rotate(${i * 3.6}deg)`,
+            },
+          })
+        );
+      };
+
+      const Hands = (timeString, fractionOfDay) => {
+        return (
+          timeString !== null &&
+          e(
+            React.Fragment,
+            null,
+            e("div", {
+              id: "short-hand",
+              style: {
+                transform: `rotate(${fractionOfDay * 360 + 180}deg)`,
+              },
+            }),
+            e("div", {
+              id: "long-hand",
+              style: {
+                transform: `rotate(${
+                  ((fractionOfDay * 10) % 1) * 360 + 180
+                }deg)`,
+              },
+            }),
+            e("div", {
+              id: "second-hand",
+              style: {
+                transform: `rotate(${
+                  ((fractionOfDay * 1000) % 1) * 360 + 180
+                }deg)`,
+              },
+            })
+          )
+        );
+      };
+
+      const Time = (timeString) => {
+        return (
+          timeString !== null &&
+          e(
+            "h1",
+            {
+              id: "clock-time",
+            },
+            timeString
+          )
+        );
+      };
+
+      const ClockFace = (timeString, fractionOfDay) => {
+        return e(
+          "div",
+          { id: "clock-face" },
+          Digits(),
+          Ticks(),
+          Hands(timeString, fractionOfDay)
+        );
+      };
+
+      const DecimalClock = () => {
+        const frameRequest = React.useRef(null);
+        const [date, setDate] = React.useState(new Date());
+        const [fractionOfDay, setFractionOfDay] = React.useState(null);
+        const [timeString, setTimeString] = React.useState(null);
+
+        const update = () => {
+          setDate(new Date());
+          frameRequest.current = requestAnimationFrame(update);
+        };
+
+        React.useEffect(() => {
+          frameRequest.current = requestAnimationFrame(update);
+
+          return () => {
+            cancelAnimationFrame(frameRequest.current);
+          };
+          // eslint-disable-next-line react-hooks/exhaustive-deps
+        }, []);
+
+        React.useEffect(() => {
+          const timeOfDay =
+            (date.getHours() * 3600 +
+              date.getMinutes() * 60 +
+              date.getSeconds()) *
+              1000 +
+            date.getMilliseconds();
+          setFractionOfDay(timeOfDay / 86400_000);
+        }, [date]);
+
+        React.useEffect(() => {
+          setTimeString(
+            `${Math.floor(fractionOfDay * 10)}:${Math.floor(
+              (fractionOfDay * 1_000) % 100
+            )
+              .toString()
+              .padStart(2, "0")}:${Math.floor((fractionOfDay * 100_000) % 100)
+              .toString()
+              .padStart(2, "0")}`
+          );
+        }, [fractionOfDay]);
```

Comment:
Calculations can be inlined. See remarks about these `useEffect` blocks in cra-implementation